test/models/ToolpadEditor.ts

```diff
@@ -108,6 +108,13 @@ export class ToolpadEditor {
     await this.createComponentDialog.createButton.click();
   }
 
+  async waitForOverlay() {
+    await this.pageOverlay.waitFor({ state: 'visible' });
+    // Some tests seem to be flaky around this waitFor and perform better with a short timeout
```

Comment:
I was removing it just to try to fix the flakiness in the tests it was causing, and also because you've said before that we should avoid test logic related to the overlay (or anything else specific to implementation) as much as possible. So I think in that sense `editorModel.pageRoot.waitFor()` might be a bit more natural?

`waitforOverlay` was making the tests work back when I first wrote them, but it has been flaky and now from what I've tested so far replacing it with `editorModel.pageRoot.waitFor()` is working just fine.

I didn't look much into the root cause of the flakiness, I thought it's probably just some specific detail on how Playwright waits for things, and the fact that we're using an iframe might make that work a bit worse than in most cases.
But if you want to keep `waitForOverlay` in case it can help us fix some bigger possible underlying issue we can do that too, and I'll just remove the instances where I replaced it.

Reply:
Over time we can probably rename it to `waitForCanvasReady` or something? I found that adding a slight delay in there does a lot to the flakeyness, although I don't like it as a solution